File: Obj-C or Swift?/tableViewObjC/tableViewObjC/ViewController.m
//
//  ViewController.m
//  tableViewObjC
//
//  Created by Vasily Bodnarchuk on 03.06.2022.
//

#import "ViewController.h"

@interface ViewController ()
@property NSString* cellIdentifier;
@property int numberOfCells;
@end

@interface ViewController (TableViewDataSource) <UITableViewDataSource>
@end

@interface ViewController (TableViewDelegate) <UITableViewDelegate>
@end

@implementation ViewController

- (id)initWithCoder:(NSCoder *)coder {
    if (self = [super initWithCoder:coder])  {
        self.cellIdentifier = @"cellIdentifier";
        self.numberOfCells = 100;
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    UITableView* tableView = [[UITableView alloc] init];
    [self.view addSubview:tableView];
    tableView.translatesAutoresizingMaskIntoConstraints = false;
    [tableView.topAnchor constraintEqualToAnchor:self.view.topAnchor].active = true;
    [tableView.leftAnchor constraintEqualToAnchor:self.view.leftAnchor].active = true;
    [tableView.rightAnchor constraintEqualToAnchor:self.view.rightAnchor].active = true;
    [tableView.bottomAnchor constraintEqualToAnchor:self.view.bottomAnchor].active = true;
    [tableView registerClass:UITableViewCell.self forCellReuseIdentifier:self.cellIdentifier];
    tableView.dataSource = self;
    tableView.delegate = self;
}
@end

// MARK: UITableViewDataSource

@implementation ViewController (TableViewDataSource)
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.numberOfCells;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
   UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier: self.cellIdentifier forIndexPath:indexPath];
    cell.textLabel.text = [NSString stringWithFormat:@"text %ld", (long)indexPath.row];
   return cell;
}
@end


// MARK: UITableViewDataSource

@implementation ViewController (TableViewDelegate)
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {

    
}
@end

File: UIKit vs SwiftUI/UIKitExample/UIKitExample/ViewController.swift
//
//  ViewController.swift
//  UIKitExample
//
//  Created by Vasily Bodnarchuk on 10.07.2022.
//

import UIKit

class ViewController: UIViewController {
    private let numberOfCells = 100
    private let cellIdentifier = "cellIdentifier"
    override func viewDidLoad() {
        super.viewDidLoad()
        let tableView = UITableView()
        view.addSubview(tableView)
        tableView.translatesAutoresizingMaskIntoConstraints = false
        tableView.topAnchor.constraint(equalTo: view.topAnchor).isActive = true
        tableView.leftAnchor.constraint(equalTo: view.leftAnchor).isActive = true
        tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
        tableView.rightAnchor.constraint(equalTo: view.rightAnchor).isActive = true
        tableView.register(UITableViewCell.self, forCellReuseIdentifier: cellIdentifier)
        tableView.dataSource = self
    }
}

// MARK: UITableViewDataSource
extension ViewController: UITableViewDataSource {
    func numberOfSections(in tableView: UITableView) -> Int {
        1
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        numberOfCells
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
        cell.textLabel?.text = "text \(indexPath.row)"
        return cell
    }
}
